client/packages/host/src/Admin/WebAppLogFileModal.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState } from 'react';
+import { useDialog, useWebClient } from '@common/hooks';
+import { useTranslation } from '@common/intl';
+import {
+  BasicSpinner,
+  DialogButton,
+  DropdownMenu,
+  DropdownMenuItem,
+  LoadingButton,
+  Typography,
+} from '@common/components';
+import { useLog } from '@openmsupply-client/system';
+import { Box, CopyIcon, SaveIcon } from 'packages/common/src';
+import { LogTextDisplay } from './LogTextDisplay';
+
+export const LogDisplay = ({
+  fileName,
+  setLogContent,
+}: {
+  fileName: string;
+  setLogContent: (content: string[]) => void;
+}) => {
+  const { data, isLoading } = useLog.document.logContentsByFileName(fileName);
+
+  if (data?.fileContent !== undefined && data?.fileContent !== null) {
+    setLogContent(data?.fileContent);
+  }
+
+  if (isLoading) {
+    return (
+      <>
+        <BasicSpinner></BasicSpinner>
+      </>
+    );
+  } else {
+    return (
+      <>
+        {Array.isArray(data?.fileContent) && data?.fileContent != undefined ? (
```

Comment:
LogTextDisplay accepts an array or string, so don't need to check if the file content is an array here